configure.ac

```diff
@@ -244,11 +244,15 @@ if test "x${GCC}" = "xyes" ; then
         else
                 AC_MSG_RESULT([< 4])
         fi
-        SHARED_FLAGS="-Wall -Wextra -Werror -ffunction-sections -fdata-sections -Wsequence-point -pipe -Wno-self-assign -Winit-self -Wmissing-include-dirs"
+        SHARED_FLAGS="-Wall -Wextra -Werror -ffunction-sections -fdata-sections -Wsequence-point -pipe -Winit-self -Wmissing-include-dirs"
```

Comment:
This commits reverts previous commit https://github.com/Mellanox/libvma/pull/161/commits/a4ed5e47b20dc0d313c7a95c6d51dd300c9f64bd